listening-platform/src/main/kotlin/com/nikitahohulia/listeningplatform/service/UserServiceImpl.kt

```diff
@@ -21,25 +21,33 @@ import reactor.kotlin.core.util.function.component2
 class UserServiceImpl(
     private val userRepository: CustomUserRepository,
     private val publisherRepository: CustomPublisherRepository,
-    private val postRepository: CustomPostRepository
+    private val postRepository: CustomPostRepository,
+    private val redisUserRepository: UserRedisRepository
 ) : UserService {
 
     override fun getUserByUsername(username: String): Mono<User> {
-        return userRepository.findByUsername(username)
-            .switchIfEmpty { NotFoundException("User not found with given username = $username").toMono() }
-    }
-
-    override fun getUserById(id: String): Mono<User> {
-        return userRepository.findById(ObjectId(id))
-            .switchIfEmpty { NotFoundException("User not found with given id = $id").toMono() }
+        return redisUserRepository.findByUsername(username)
+            .switchIfEmpty {
+                userRepository.findByUsername(username)
+                    .flatMap { user ->
+                        redisUserRepository.save(user).thenReturn(user)
+                    }
+                    .switchIfEmpty(NotFoundException(message = "User not found with username = $username").toMono())
+            }
     }
 
     override fun createUser(user: User): Mono<User> {
-        return userRepository.findByUsername(user.username)
+        return redisUserRepository.findByUsername(user.username)
+            .switchIfEmpty {
+                userRepository.findByUsername(user.username)
+            }
```

Comment:
```kotlin
redisUserRepository.findByUsername(user.username)
            .switchIfEmpty {
                userRepository.findByUsername(user.username)
            }
```

This code repeated multiple times. Maybe it's time to extract a private function?

Reply:
Fixed that in other way